eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,508 @@
+              parameters:
+                Paths:
+                  - '${{ parameters.ServiceDirectory }}'
+                  - '**/*.xml'
+                  - '!sdk/**/test-recordings'
+                  - '!sdk/**/session-records'
+
+            - task: UsePythonVersion@0
+              displayName: 'Use Python 3.6'
+              inputs:
+                versionSpec: '3.6'
+
+            - template: /eng/common/pipelines/templates/steps/set-test-pipeline-version.yml
+              parameters:
+                PackageName: "azure-sdk-template"
+                ServiceDirectory: "template"
+                TestPipeline: ${{ parameters.TestPipeline }}
+
+            - template: /eng/common/pipelines/templates/steps/set-test-pipeline-version.yml
+              parameters:
+                PackageName: "azure-sdk-template-two"
+                ServiceDirectory: "template"
+                TestPipeline: ${{ parameters.TestPipeline }}
+
+            - template: /eng/common/pipelines/templates/steps/set-test-pipeline-version.yml
+              parameters:
+                PackageName: "azure-sdk-template-three"
+                ServiceDirectory: "template"
+                TestPipeline: ${{ parameters.TestPipeline }}
+
+            # Do all the verifications for all the artifacts after the sparce-checkout
+            - pwsh: |
+                $ArtifactsList = $('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object name, groupId)
+                $ArtifactsList | Format-Table -Property GroupId, Name | Out-String | Write-Host
+                python --version
+                foreach($artifact in $ArtifactsList) {
+                  python eng/versioning/set_versions.py --build-type ${{parameters.SDKType}} --vv --ai $($artifact.name) --gi $($artifact.groupId)
+                }
+              displayName: 'Verify artifact versions for release'
+
+            # Uber JARs release a flattened POM, as dependencies are incorporated into the JAR circumnavigating the
+            # need to list dependencies in the POM. The flattened POM is generated and won't comply with our versioning
+            # validation, so omit uberJars from the artifacts list to verify.
+            - pwsh: |
+                $ArtifactsList = $('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object name, groupId, uberJar | Where-Object -Not "uberJar")
+                $ArtifactsList | Format-Table -Property GroupId, Name | Out-String | Write-Host
+                python --version
+                foreach($artifact in $ArtifactsList) {
+                  $(Build.SourcesDirectory)/eng/versioning/scan_for_unreleased_dependencies.ps1 $($artifact.groupId) $($artifact.name) $(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}
+                }
+              displayName: "Verify no unreleased dependencies or beta dependencies for GA libraries"
+
+            - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
[in follow-up] we should improve this to verify them as a batch but it will require changes to eng/common.